packages/babel-plugin-transform-class-properties/src/index.js

```diff
@@ -131,11 +131,17 @@ export default function ({ types: t }) {
         if (!nodes.length) return;
 
         if (path.isClassExpression()) {
-          nodes.push(t.expressionStatement(ref));
-        }
-
-        if (path.isClassDeclaration() && path.parentPath.isExportDeclaration()) {
+          path.scope.push({ id: ref });
+          path.replaceWith(t.assignmentExpression("=", ref, path.node));
+        } else if (path.parentPath.isExportDeclaration()) {
+          let node = path.node;
           path = path.parentPath;
+
+          if (path.isExportDefaultDeclaration() && !node.id) {
+            node.type = "ClassExpression";
+            path.scope.push({ id: ref });
+            path.replaceWith(t.exportDefaultDeclaration(t.assignmentExpression("=", ref, node)));
```

Comment:
I think this will introduce a similar issue to the one I fixed in https://github.com/babel/babel/pull/3096.

Can't we assign `node.id = ref` to ensure the class has an identifier instead of changing the export declarations like this?